pkg/operator/ceph/object/controller_test.go

```diff
@@ -655,3 +662,250 @@ func TestCephObjectStoreControllerMultisite(t *testing.T) {
+		{Name: "my-store.rgw.buckets.data"},
+		{Name: ".rgw.root"},
+		{Name: "my-store.rgw.buckets.index"},
+	}
+
+	osdGetPoolOutput := func(poolName string) string {
+		return fmt.Sprintf(`{
+	"pool": "%s",
+	"pool_id": 5,
+	"size": 1,
+	"min_size": 1,
+	"pg_num": 8,
+	"pgp_num": 8,
+	"crush_rule": "%s",
+	"hashpspool": true,
+	"nodelete": false,
+	"nopgchange": false,
+	"nosizechange": false,
+	"write_fadvise_dontneed": false,
+	"noscrub": false,
+	"nodeep-scrub": false,
+	"use_gmt_hitset": true,
+	"fast_read": 0,
+	"compression_mode": "none",
+	"pg_autoscale_mode": "on",
+	"pg_num_min": 8
+}`, poolName, poolName)
+	}
+
+	rgwAdminRealmGetOutput := `{
+	"id": "29e44350-b208-4f25-a3ce-6a059ea559f2",
+	"name": "my-store",
+	"current_period": "0bd7f624-a22e-4779-a64d-913824d7e843",
+	"epoch": 2
+}`
+
+	rgwAdminZoneGroupGetOutput := `{
+	"id": "e03a68ee-f4a0-438c-826a-399ace923f2f",
+	"name": "my-store",
+	"api_name": "my-store",
+	"is_master": "true",
+	"endpoints": [
+		"http://10.111.19.44:80"
+	],
+	"hostnames": [],
+	"hostnames_s3website": [],
+	"master_zone": "5d4f36ea-b4b4-405c-8b60-0c7bfc71072f",
+	"zones": [
+		{
+			"id": "5d4f36ea-b4b4-405c-8b60-0c7bfc71072f",
+			"name": "my-store",
+			"endpoints": [
+				"http://10.111.19.44:80"
+			],
+			"log_meta": "false",
+			"log_data": "false",
+			"bucket_index_max_shards": 11,
+			"read_only": "false",
+			"tier_type": "",
+			"sync_from_all": "true",
+			"sync_from": [],
+			"redirect_zone": ""
+		}
+	],
+	"placement_targets": [
+		{
+			"name": "default-placement",
+			"tags": [],
+			"storage_classes": [
+				"STANDARD"
+			]
+		}
+	],
+	"default_placement": "default-placement",
+	"realm_id": "29e44350-b208-4f25-a3ce-6a059ea559f2",
+	"sync_policy": {
+		"groups": []
+	}
+}`
+
+	rgwAdminZoneGetOutput := `{
+	"id": "5d4f36ea-b4b4-405c-8b60-0c7bfc71072f",
+	"name": "my-store",
+	"domain_root": "my-store.rgw.meta:root",
+	"control_pool": "my-store.rgw.control",
+	"gc_pool": "my-store.rgw.log:gc",
+	"lc_pool": "my-store.rgw.log:lc",
+	"log_pool": "my-store.rgw.log",
+	"intent_log_pool": "my-store.rgw.log:intent",
+	"usage_log_pool": "my-store.rgw.log:usage",
+	"roles_pool": "my-store.rgw.meta:roles",
+	"reshard_pool": "my-store.rgw.log:reshard",
+	"user_keys_pool": "my-store.rgw.meta:users.keys",
+	"user_email_pool": "my-store.rgw.meta:users.email",
+	"user_swift_pool": "my-store.rgw.meta:users.swift",
+	"user_uid_pool": "my-store.rgw.meta:users.uid",
+	"otp_pool": "my-store.rgw.otp",
+	"system_key": {
+		"access_key": "",
+		"secret_key": ""
+	},
+	"placement_pools": [
+		{
+			"key": "default-placement",
+			"val": {
+				"index_pool": "my-store.rgw.buckets.index",
+				"storage_classes": {
+					"STANDARD": {
+						"data_pool": "my-store.rgw.buckets.data"
+					}
+				},
+				"data_extra_pool": "my-store.rgw.buckets.non-ec",
+				"index_type": 0
+			}
+		}
+	],
+	"realm_id": "",
+	"notif_pool": "my-store.rgw.log:notif"
+}`
+
+	t.Run("health checker fails to be created", func(t *testing.T) {
```

Comment:
A lot of new boilerplate just to check that failure to start the health checker will fail the reconcile and cause it to run again.